platform-core-module/src/main/java/com/softicar/platform/core/module/file/stored/content/store/StoredFileSmbContentStore.java

```diff
@@ -160,50 +113,26 @@ public long getFileSize(String filename) {
 	@Override
 	public Collection<String> getAllFiles(String root) {
 
-		return listAllFiles(createSmbFile(root), Trim.trimRight(root, '/'), new ArrayList<>());
+		return createSmbFile(root)//
+			.asDirectory()
+			.orElseThrow()
+			.listFilesRecursively(Trim.trimRight(root, '/'));
 	}
 
 	@Override
 	public DayTime getLastModified(String filename) {
 
-		SmbFile file = createSmbFile(filename);
-
-		try {
-			return DayTime.fromDate(new Date(file.lastModified()));
-		} catch (SmbException exception) {
-			throw new SofticarIOException(exception);
-		}
-	}
-
-	private Collection<String> listAllFiles(SmbFile root, String prefix, Collection<String> filenames) {
-
-		try {
-			for (SmbFile file: root.listFiles()) {
-				if (file.isDirectory()) {
-					String subFolder = Trim.trimRight(file.getName(), '/');
-					listAllFiles(file, prefix + "/" + subFolder, filenames);
-				} else {
-					filenames.add(prefix + "/" + file.getName());
-				}
-			}
-		} catch (SmbException exception) {
-			throw new SofticarException(exception);
-		}
-
-		return filenames;
-	}
-
-	private SmbFile createSmbFile(String name) {
-
-		return JcifsSmbFileUtils.createFile(getServerOrThrow().getUrl() + "/" + name, getAuthentication(getServerOrThrow()));
+		return createSmbFile(filename).getLastModifiedDate();
 	}
 
-	private NtlmPasswordAuthentication getAuthentication(AGStoredFileServer fileServer) {
+	private ISmbFile createSmbFile(String name) {
 
-		return new NtlmPasswordAuthentication(//
-			fileServer.getDomain(),
-			fileServer.getUsername(),
-			fileServer.getPassword());
+		AGStoredFileServer fileServer = getServerOrThrow();
+		return CurrentSmbClient
+			.get()
+			.createFile(
+				getServerOrThrow().getUrl() + "/" + name,
```

Comment:
```suggestion
				Trim.trimRight(getServerOrThrow().getUrl(), '/') + "/" + name,
```
We should probably remove a tailing `/` from the return value of `getUrl`, it any, to make sure that we never end up with `//` in the resulting URL.